### PR TITLE
list terms

### LINKED ISSUE #2567: List of terms for field
Given a field `f`, a prefix `p`, and a maximum number of returned terms `n`, returns at most `n` unique terms ordered lexicographically starting with the prefix `p` for the field `f`. Returns all the terms in `f` if `p` is the empty string and `n` is `None`.

Use case: for the Jaeger integration, we want to efficiently retrieve the list of `services` for which traces have been ingested in the last 24h. Since the term dictionary will be small for this field, we will likely want to store it in the hotcache.

### PR DESCRIPTION
**Pull request by trinity-1686a**
fix #2567 

### Description

add a new list_term search, which enumerate terms over a range in the sstable and *does not* fetch actual documents

### How was this PR tested?

basic unit tests where added

### TODO
error handling is perfectible/missing, some code (term manipulation) should probably be part of tantivy rather than quickwit

### CODE REVIEW

File: quickwit/quickwit-proto/protos/quickwit/search_api.proto [hunk 1]
```diff
@@ -44,6 +44,21 @@ service SearchService {
 
   // Perform a leaf stream on a given set of splits.
   rpc LeafSearchStream(LeafSearchStreamRequest) returns (stream LeafSearchStreamResponse);
+
+  // Root list term API.
```
Comment (guilload) [inline quickwit/quickwit-proto/protos/quickwit/search_api.proto:48]:
```suggestion
  // Root list terms API.
```

File: quickwit/quickwit-search/src/collector.rs [hunk 2]
```diff
@@ -110,15 +110,12 @@ fn resolve_sort_by(
...
-            if let Some(field) = segment_reader.schema().get_field(field_name) {
-                let fast_field_reader = segment_reader.fast_fields().u64_lenient(field)?;
-                Ok(SortingFieldComputer::FastField {
-                    fast_field_reader,
-                    order: *order,
-                })
-            } else {
-                Ok(SortingFieldComputer::DocId)
-            }
+            // /!\ behavior change if field_name does not exists
```
Comment (guilload) [inline quickwit/quickwit-search/src/collector.rs:113]: That's for the best.

**Review by guilload — APPROVED** [review]

File: quickwit/quickwit-search/src/collector.rs (hunk 2 above)
Comment (trinity-1686a) [reply quickwit/quickwit-search/src/collector.rs:113]: So can I remove that comment?

Comment (guilload) [reply quickwit/quickwit-search/src/collector.rs:113]: Yes.

File: quickwit/quickwit-proto/protos/quickwit/search_api.proto [hunk 3]
```diff
@@ -44,6 +44,21 @@ service SearchService {
 
   // Perform a leaf stream on a given set of splits.
   rpc LeafSearchStream(LeafSearchStreamRequest) returns (stream LeafSearchStreamResponse);
+
+  // Root list terms API.
+  // This RPC identifies the set of splits on which the query should run on,
+  // and dispatches the several calls to `LeafListTerm`.
```
Comment (guilload) [inline quickwit/quickwit-proto/protos/quickwit/search_api.proto:50]:
```suggestion
  // and dispatches the several calls to `LeafListTerms `.
```

File: quickwit/quickwit-search/src/cluster_client.rs [hunk 4]
```diff
@@ -144,6 +144,32 @@ impl ClusterClient {
...
     }
+
+    /// Leaf search with retry on another node client.
+    pub async fn leaf_list_terms(
+        &self,
+        request: LeafListTermsRequest,
+        mut client: SearchServiceClient,
+    ) -> crate::Result<LeafListTermsResponse> {
+        let response_res = client.leaf_list_terms(request.clone()).await;
+        // TODO: implement retry
```
Comment (guilload) [inline quickwit/quickwit-search/src/cluster_client.rs:155]: Can you remove this commented code and open an issue?

**Review by guilload — APPROVED** [review]

**Pull request merged**